src/twisted/conch/ssh/transport.py

```diff
@@ -730,34 +730,56 @@ def dataReceived(self, data):
         self.buf = self.buf + data
         if not self.gotVersion:
 
-            if len(self.buf) > 4096:
+            if len(self.buf) >= 255 and b"\n" not in self.buf:
+                # Looks like we have already received enough data,
+                # yet it doesn't look like the version string line
+                # is in the received data.
+                # RFC 4253 section 4.2
+                # The maximum length of the string is 255 characters,
+                # including the Carriage Return and Line Feed.
                 self.sendDisconnect(
                     DISCONNECT_CONNECTION_LOST,
-                    b"Peer version string longer than 4KB. "
-                    b"Preventing a denial of service attack.",
+                    b"Invalid peer version. Long value. Preventing a denial of service attack.",
```

Comment:
```suggestion
                    b"Peer identification string not found in first 255 characters.",
```

All other things being equal I prefer using the "identification string" language from the RFC.  I don't object to the DoS attack language, but it seems unnecessary - I'm sure there are various errors that serve the purpose of preventing DoS attacks, and we probably shouldn't imply in the logs that only this one does so.

Reply:
Agree. I will use your suggestion. Agree that DoS is not needed. Thanks!